Orange/widgets/utils/signals.py

```diff
@@ -0,0 +1,242 @@
+import copy
+
+from Orange.canvas.registry.description import InputSignal, OutputSignal
+
+
+class _Signal:
+    @staticmethod
+    def get_flags(multiple, default, explicit, dynamic):
+        """Compute flags from arguments"""
+        from Orange.canvas.registry.description import \
+            Single, Multiple, Default, NonDefault, Explicit, Dynamic
+        return (Multiple if multiple else Single) | \
+                (Default if default else NonDefault) | \
+                (explicit and Explicit) | \
+                (dynamic and Dynamic)
+
+
+class Input(InputSignal, _Signal):
+    """
+    Description of an input signal.
+
+    The class is used to declare input signals for a widget as follows
+    (the example is taken from the widget Test & Score)::
+
+        class Inputs:
```

Comment:
I don't think it would clash, it's a different namespace.

I think the plural looks way better. We're listing outputs here. Having singular in messages was a mistake, I think. I used it so that raising messages looked better (`self.Error.data_mismatch()`), but I often make a mistake of declaring the class as `class Errors` instead of `class Error`.

Now, do we
- make the same mistake here, for consistency,
- use plural here, ignoring the consistency (but, heck, is plural indeed so much better?),
- change messages to plural (but see the comment above)?